starknet_py/tests/e2e/deploy/deployer_test.py

```diff
@@ -94,6 +94,7 @@ async def test_constructor_arguments_contract_deploy(
     )
 
 
+# TODO when changed to `account`, tests (1, True) and (2, False) fail on full_node_account
```

Comment:
Could you perform some further investigation?
Is this a bug is FullNodeClient or?
What's the error message?

Reply:
In both cases, the transaction gets rejected with an unknown error, so I don't think it is an issue with FullNodeClient
```
salt = 1, pass_account_address = True
account = <starknet_py.net.account.account.Account object at 0x116a7e250>
map_class_hash = 269454390811152316...6789178223652524401

    @pytest.mark.asyncio
    @pytest.mark.parametrize(
        "salt, pass_account_address", [(1, True), (2, False), (None, True), (None, False)]
    )
    async def test_address_computation(
        salt, pass_account_address, account, map_class_hash
    ):
        deployer = Deployer(
            account_address=account.address if pass_account_address else None
        )
    
        deploy_call, computed_address = deployer.create_deployment_call(
            class_hash=map_class_hash, salt=salt
        )
    
        deploy_invoke_tx = await account.sign_invoke_transaction(
            deploy_call, max_fee=MAX_FEE
        )
        resp = await account.client.send_transaction(deploy_invoke_tx)
>       await account.client.wait_for_tx(resp.transaction_hash)

deployer_test.py:117: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <starknet_py.net.full_node_client.FullNodeClient object at 0x116a7edc0>
tx_hash = 159922644870949472...4208212143205555769, wait_for_accept = False
check_interval = 5

    async def wait_for_tx(
        self,
        tx_hash: Hash,
        wait_for_accept: Optional[bool] = False,
        check_interval=5,
    ) -> Tuple[int, TransactionStatus]:
        # pylint: disable=too-many-branches
        """
        Awaits for transaction to get accepted or at least pending by polling its status
    
        :param tx_hash: Transaction's hash
        :param wait_for_accept: If true waits for at least ACCEPTED_ON_L2 status, otherwise waits for at least PENDING
        :param check_interval: Defines interval between checks
        :return: Tuple containing block number and transaction status
        """
        if check_interval <= 0:
            raise ValueError("Argument check_interval has to be greater than 0.")
    
        first_run = True
        try:
            while True:
                result = await self.get_transaction_receipt(tx_hash=tx_hash)
                status = result.status
    
                if status in (
                    TransactionStatus.ACCEPTED_ON_L1,
                    TransactionStatus.ACCEPTED_ON_L2,
                ):
                    assert result.block_number is not None
                    return result.block_number, status
                if status == TransactionStatus.PENDING:
                    if not wait_for_accept:
                        if result.block_number is not None:
                            return result.block_number, status
                elif status == TransactionStatus.REJECTED:
>                   raise TransactionRejectedError(
                        message=result.rejection_reason,
E                       starknet_py.transaction_exceptions.TransactionRejectedError: Transaction was rejected with following Starknet error: Unknown Starknet error..

../../../net/client.py:169: TransactionRejectedError
```

Reply:
Something similar was happening in #975 which has been fixed by altering the constructor arguments.

I'd suggest try changing something between the set of tests for GatewayClient and FullNodeClient (map_contract does not have a constructor, so maybe try chaning the max_fee? if that does not work maybe try with a contract that has a constructor and do something similar to #975).

Not exactly sure what causes the problem atm.
We probably would want to investigate this further, might be a devnet issue.

Reply:
Salt is now changed when`full_node_account` is tested, transactions with identical parameters won't be accepted I guess.